src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -54,8 +54,15 @@ public function loadUserByUsername($username)
         if (null !== $this->property) {
             $user = $this->repository->findOneBy(array($this->property => $username));
         } else {
-            if (!$this->repository instanceof UserProviderInterface) {
-                throw new \InvalidArgumentException(sprintf('The Doctrine repository "%s" must implement UserProviderInterface.', get_class($this->repository)));
+            if (!$this->repository instanceof UserLoaderInterface) {
```

Comment:
I rephrased to clause a bit as the deprecation warning should be thrown if the `UserLoaderInterface` is not being implemented.